File: Android/callPlugin/assets/www/js/callplugin.js
/*
 * CallPlugin
 * Plugin per a Phonegap 2.0
 * Automatitza trucades de telèfon
 * 
 * (c) 2012, Josep Lluis Monte Galiano
 * @jlmoga
 * www.moga.cat
 * moga@moga.cat
 */

var CallPlugin = (function (gap) {
    function isFunction(f) {
        return typeof f === "function";
    }

    // CallPlugin
    function CallPlugin() {}

    
    /**
     * Fa una trucada al telèfon especificat
     *
     * @param phone   Telèfom
     */
    CallPlugin.call = function (phone) {
        gap.exec(onEvent, onError, "CallPlugin", "Call", [phone]);
    };

    
    /**
     * Esdeveniment en cas d'èxit. La trucada es fa.
     */
    function onEvent(data) {
        CallPlugin.onSuccess();
    }

    /**
     * Esdeveniment en cas d'error.
     */
    function onError(data) {
        CallPlugin.onError(data);
    }

    
    /**
     * Càrrega CallPlugin
     */
    gap.addConstructor(function () {
        if (gap.addPlugin) {
            gap.addPlugin("callplugin", CallPlugin);
        } else {
            if (!window.plugins) {
                window.plugins = {};
            }

            window.plugins.callPlugin = CallPlugin;
        }
    });

    return CallPlugin;
})(window.cordova || window.Cordova || window.PhoneGap);

File: Android/callPlugin/src/com/moga/plugins/callplugin/CallPlugin.java
/*
 * CallPlugin
 * Plugin per a Phonegap 2.0
 * Automatitza trucades de telèfon
 * 
 * (c) 2012, Josep Lluis Monte Galiano
 * @jlmoga
 * www.moga.cat
 * moga@moga.cat
 */

package com.moga.plugins.callplugin;

import org.apache.cordova.api.PluginResult.Status;
import org.json.JSONArray;
import org.json.JSONException;

import android.net.Uri;

import android.content.Intent;

import org.apache.cordova.api.Plugin;
import org.apache.cordova.api.PluginResult;

public class CallPlugin extends Plugin {
	public final String ACTION_CALL = "Call";
	
	/**
	 * Automatització del Plugin
	 * 
	 * @param action  Acció a executar
	 * @param argl    Arguments de l'acció
	 */
	@Override
	public PluginResult execute(String action, JSONArray arg1, String callbackId) {
	    PluginResult result = new PluginResult(Status.INVALID_ACTION);

	    try {
	    	if (action.equals(ACTION_CALL)) {
	            String number = "tel:" + arg1.getString(0);
				call(number);
				result = new PluginResult(Status.OK);
	        }
	    } catch (JSONException ex) {
	    	result = new PluginResult(Status.JSON_EXCEPTION, ex.getMessage());
	    }

	    return result;
	}

    /**
     * Fa una trucada al telèfon especificat
     *
     * @param phoneNumber   Telèfom
     */
	private void call(String phoneNumber) {
	    Intent callIntent = new Intent(Intent.ACTION_CALL, Uri.parse(phoneNumber)); 
        this.cordova.getActivity().startActivity(callIntent);	
	}
}
